front/lib/resources/data_source_view_resource.ts

```diff
@@ -539,6 +539,24 @@ export class DataSourceViewResource extends ResourceWithSpace<DataSourceViewMode
     return getDataSourceViewUsage({ auth, dataSourceView: this });
   };
 
+  // Permissions.
+
+  /**
+   * Determines if the current user has write permissions.
+   * For managed data sources, requires administrative privileges.
+   * For non-managed data sources, requires write privileges.
+   *
+   * @param auth - The authenticator object for the current user
+   * @returns boolean indicating whether the user has write permission
+   */
+  canWrite(auth: Authenticator) {
+    if (isManaged(this.dataSource)) {
+      return this.space.canAdministrate(auth);
+    }
```

Comment:
I'd rather avoid an if here that bypass the permission system.

What about updating for restricted space, from:

`roles: [{ role: "admin", permissions: ["admin"] }]`

to

`roles: [{ role: "admin", permissions: ["admin", "write"] }],`

(we do it for public space)

Reply:
in `requestedPermissions()`

Reply:
Wanted to go with this at first, but we don't want admins to be able to write other data sources than connected. This being said, I'm just gonna go for it, since it makes the whole thing way easier 😅